src/gui/folderstatusmodel.cpp

```diff
@@ -1090,10 +1090,7 @@ void FolderStatusModel::slotFolderSyncStateChange(Folder *f)
     if (f->syncResult().folderStructureWasChanged()
         && (state == SyncResult::Success || state == SyncResult::Problem)) {
         // There is a new or a removed folder. reset all data
-        auto &info = _folders[folderIndex];
-        auto idx = index(folderIndex);
-        info.resetSubs(this, idx);
-        fetchMore(idx);
+        fetchMore(index(folderIndex));
```

Comment:
this does nothing if the older was already fetched. 
We need to clean the cache or do something so we are sure to actually do a request and remove the folders that no longer exist.

Reply:
I don't get it.